test/factory/smartWalletFactory.test.ts

```diff
@@ -1,497 +1,557 @@
-//                 )
-//             ).to.be.rejectedWith(
-//                 'Returned error: VM Exception while processing transaction: revert nonce mismatch'
-//             );
-//         });
-
-//         it('Should fail when signature does not match', async () => {
-//             const relayRequest = createRequest(
-//                 {
-//                     from: owner.address,
-//                     to: logicAddress,
-//                     data: initParams,
-//                     tokenContract: token.address,
-//                     tokenAmount: '5000',
-//                     tokenGas: '0',
-//                     index: index,
-//                     relayHub: worker
-//                 },
-//                 {
-//                     callForwarder: factory.address
-//                 }
-//             );
-
-//             const { signature, suffixData } = signRequest(
-//                 owner.privateKey,
-//                 relayRequest,
-//                 chainId
-//             );
-
-//             relayRequest.request.from = otherAccount;
-
-//             await expect(
-//                 factory.relayedUserSmartWalletCreation(
-//                     relayRequest.request,
-//                     suffixData,
-//                     worker,
-//                     signature,
-//                     {
-//                         from: worker
-//                     }
-//                 )
-//             ).to.be.rejectedWith(
-//                 'Returned error: VM Exception while processing transaction: revert Signature mismatch'
-//             );
-//         });
-//     });
-// });
+import { expect } from 'chai';
+import { SmartWallet, SmartWalletFactory, UtilToken } from 'typechain-types';
+import { ethers } from 'hardhat';
+import { constants, utils, Wallet } from 'ethers';
+import { createValidPersonalSignSignature } from '../utils/createValidPersonalSignSignature';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { createDeployRequest } from './utils';
+import {
+  getLocalEip712DeploySignature,
+  TypedDeployRequestData,
+} from '../utils/EIP712Utils';
+import { getSuffixData, HARDHAT_CHAIN_ID } from '../smartwallet/utils';
+
+const minIndex = 0;
+const maxIndex = 1000000000;
+
+const nextIndex = () =>
+  Math.floor(Math.random() * (maxIndex - minIndex + 1) + minIndex);
+
+const deployContract = <T>(contract: string) => {
+  return ethers
+    .getContractFactory(contract)
+    .then((contractFactory) => contractFactory.deploy() as T);
+};
+
+const provider = ethers.provider;
+
+describe('SmartWalletFactory', function () {
+  describe('constructor', function () {
+    let smartWalletFactory: SmartWalletFactory;
+    let template: Wallet;
+
+    beforeEach(async function () {
+      template = ethers.Wallet.createRandom();
+      const factory = await ethers.getContractFactory('SmartWalletFactory');
+      smartWalletFactory = await factory.deploy(template.address);
```

Comment:
Why don't we use the `deployContract` function here? if it's because of the arguments, we could use the `deployContract` function defined in `deployment.utils.ts`

Reply:
updated